src/Program.ts

```diff
@@ -385,16 +402,20 @@ export class Program {
      * @param xmlFile XML file with <component> tag
      */
     private addDeferredComponentTypeSymbolCreation(xmlFile: XmlFile) {
-        this.componentSymbolsToUpdate.add({ componentKey: this.getComponentKey(xmlFile), componentName: xmlFile.componentName?.text });
-
+        const componentKey = this.getComponentKey(xmlFile);
```

Comment:
changed this to a map with unique keys, because there multiple places where it can get updated